src/lib/services/client-metrics/index.ts

```diff
@@ -1,54 +1,148 @@
 /* eslint-disable no-param-reassign */
+import EventStore, { ICreateEvent } from '../../db/event-store';
+import StrategyStore from '../../db/strategy-store';
+import ClientApplicationsDb from '../../db/client-applications-store';
+import ClientInstanceStore from '../../db/client-instance-store';
+import { ClientMetricsStore } from '../../db/client-metrics-store';
+import FeatureToggleStore from '../../db/feature-toggle-store';
+import { LogProvider } from '../../logger';
+import { applicationSchema } from './metrics-schema';
+import { Projection } from './projection';
+import { clientMetricsSchema } from './client-metrics-schema';
+import { APPLICATION_CREATED } from '../../types/events';
+import { IApplication, IYesNoCount } from './models';
+import { IUnleashStores } from '../../types/stores';
+import { IUnleashConfig } from '../../types/option';
 
-'use strict';
-
-const Projection = require('./projection');
 const TTLList = require('./ttl-list');
-const appSchema = require('./metrics-schema');
-const { clientMetricsSchema } = require('./client-metrics-schema');
 const { clientRegisterSchema } = require('./register-schema');
-const { APPLICATION_CREATED } = require('../../types/events');
 
 const FIVE_SECONDS = 5 * 1000;
 const FIVE_MINUTES = 5 * 60 * 1000;
 
-module.exports = class ClientMetricsService {
+export interface IClientApp {
+    appName: string;
+    instanceId: string;
+    clientIp?: string;
+    seenToggles?: string[];
+    metricsCount?: number;
+    strategies?: string[] | Record<string, string>[];
+    bucket?: any;
+    count?: number;
+    started?: number | Date;
+    interval?: number;
+    icon?: string;
+    description?: string;
+    color?: string;
+}
+
+export interface IAppFeature {
+    name: string;
+    description: string;
+    type: string;
+    project: string;
+    enabled: boolean;
+    stale: boolean;
+    strategies: any;
+    variants: any[];
+    createdAt: Date;
+    lastSeenAt: Date;
+}
```

Comment:
heads up @chriswk here we get a definition of a feature toggle that will be changed in upcomming envrionments features [on branch](https://github.com/Unleash/unleash/tree/wip/environments)